src/command_line.ts

```diff
@@ -52,26 +44,65 @@ export class CommandLine {
    * Construct an object of this class.
    *
    * @param denoArgs - See this.deno_args property.
-   * @param subcommands - Used to match subcommand arguments to arguments in the
-   * command line.
    */
-  constructor(denoArgs: string[], subcommands: Subcommand[]) {
-    this.deno_args = denoArgs;
-
-    // The second argument is always the subcommand
-    this.subcommand = this.deno_args.shift() as string;
-
-    this.extractDenoFlagsFromArguments();
-
-    this.extractOptionsFromArguments();
-
-    this.matchArgumentsToNames(subcommands);
+  constructor(cli: Line.Cli, denoArgs: string[] = []) {
+    this.cli = cli;
+    this.deno_args = denoArgs.slice(); // Make a copy so they can be mutated
+    this.formatCommandLine();
   }
 
   //////////////////////////////////////////////////////////////////////////////
   // FILE MARKER - METHODS - PUBLIC ////////////////////////////////////////////
   //////////////////////////////////////////////////////////////////////////////
 
+  /**
+   * Extract all options and their values from Deno.args.
+   */
+  public extractOptionsFromArguments(
+    command: Line.Command | Line.Subcommand,
+  ): void {
+    this.deno_args.forEach((datum: string) => {
+      if (datum.match(/^-/)) {
+        this.options[datum] = true; // Default value is true
+      }
+    });
+
+    for (const optionName in this.options) {
+      // Get the location of the option in the line
+      const index = this.deno_args.indexOf(optionName);
+
+      // The input AFTER the location of the option is the value of the option
+      // UNLESS it is also another option. If the input after the location of
+      // the option is another option, then all we do is say, "Yes, the option
+      // exists in the command line," and we just assign it a `true` value.
+      let value: string | boolean = this.deno_args[index + 1];
+
+      const isOption = command.options_parsed.indexOf(value) != -1 ||
+        (value in this.options);
+
+      if (isOption) {
+        value = true;
+      }
+
+      // If the option is last in the deno args and has no value, then we do not
+      // want its value to be `undefined`. We want its value to be `true`
+      // because it exists in the command line.
+      if (this.deno_args.length <= (index + 1)) {
+        value = true;
+      }
+
+      if (value == "true") {
+        value = true;
+      }
+
+      if (value == "false") {
+        value = false;
+      }
```

Comment:
what about doing a `value = Boolean(value)`?

Reply:
for the life of me i couldn't figure out why this code looked so jank. thank you